ocf_datapipes/transform/xarray/pv/create_pv_image.py

```diff
@@ -42,107 +44,125 @@ def __init__(
+                elif pv_coords == "lat_lon" and image_coords == "geostationary":
+                    pv_x, pv_y = lon_lat_to_geostationary_area_coords(
+                        x=pv_system["longitude"].values,
+                        y=pv_system["latitude"].values,
+                        xr_data=image_xr,
                     )
+
                 else:
-                    pv_x = pv_system["x_osgb"]
-                    pv_y = pv_system["y_osgb"]
-                # Quick check as search sorted doesn't give an error if it is not in the range
-                if pv_x < image_xr[self.x_dim][0].values or pv_x > image_xr[self.x_dim][-1].values:
-                    continue
-                # Y Coordinates are in reverse for satellite data
-                if pv_y > image_xr[self.y_dim][0].values or pv_y < image_xr[self.y_dim][-1].values:
-                    continue
-                if "geostationary" in self.x_dim:
-                    x_idx = np.searchsorted(image_xr[self.x_dim].values, pv_x) - 1
-                    # y_geostationary is in descending order:
-                    y_idx = len(image_xr[self.y_dim]) - (
-                        np.searchsorted(image_xr[self.y_dim].values[::-1], pv_y) - 1
+                    raise NotImplementedError(
+                        f"PV coords of type {pv_coords} and image coords of type {image_coords}"
                     )
-                else:
-                    x_idx = np.searchsorted(pv_x, image_xr[self.x_dim])
-                    y_idx = np.searchsorted(pv_y, image_xr[self.y_dim])
-                if x_idx == len(image_xr[self.x_dim]) or y_idx == len(image_xr[self.y_dim]):
+
+                # Check the PV system is within the image
+                in_range = (image_xr[image_x_dim].min() < pv_x < image_xr[image_x_dim].max()) and (
+                    image_xr[image_y_dim].min() < pv_y < image_xr[image_y_dim].max()
+                )
+
+                # If normalizing by pvlib we need tilt and orientation values
+                is_normalizable = (not self.normalize_by_pvlib) or np.isfinite(
+                    [pv_system.orientation.values, pv_system.tilt.values]
+                ).all()
+
+                if not in_range or not is_normalizable:
+                    # Skip the PV system if not inside the image
+                    # Skip the PV system if pvlib normalizarion requested but cannot be completed
                     continue
-                # Add location to same one, so know if multiple overlap
-                # Filter if normalizing by pvlib,
-                # only include ones that have tilt and orientation as numbers
-                if self.normalize_by_pvlib:
-                    if not np.isfinite(pv_system.orientation.values) and not np.isfinite(
-                        pv_system.tilt.values
-                    ):
-                        continue
+
+                x_idx = searchsorted(image_xr[image_x_dim], pv_x, assume_ascending=x_ascend)
```

Comment:
@jacobbieker I might have messed this up. I think it should be something like 
```
x_idx = searchsorted(image_xr[image_x_dim], pv_x, assume_ascending=x_ascend) - 1
```
to match what it was before. 

But maybe we should use something like :
```
 x_idx = image_xr.get_index(image_x_dim).get_indexer([pv_x], method="nearest")[0]
```
This means pick the nearest index for each PV system rather than rounding up (the code as it is now) or down (as the change above)

Reply:
Yeah, getting the nearest one is probably best.